l1-contracts/src/core/libraries/crypto/SampleLib.sol

```diff
@@ -32,203 +41,67 @@ library SampleLib {
    */
-  function computeCommitteeStupid(uint256 _committeeSize, uint256 _indexCount, uint256 _seed)
+  function computeCommittee(uint256 _committeeSize, uint256 _indexCount, uint256 _seed)
     internal
-    pure
     returns (uint256[] memory)
   {
-    uint256[] memory indices = new uint256[](_committeeSize);
+    require(_committeeSize <= _indexCount, Errors.SampleLib__SampleLargerThanIndex(_committeeSize, _indexCount));
 
-    for (uint256 index = 0; index < _indexCount; index++) {
-      uint256 sampledIndex = computeShuffledIndex(index, _indexCount, _seed);
-      if (sampledIndex < _committeeSize) {
-        indices[sampledIndex] = index;
-      }
-    }
+    uint256[] memory sampledIndices = new uint256[](_committeeSize);
 
-    return indices;
-  }
-
-  /**
-   * @notice  Computing a committee slightly more cleverly.
-   *          Only computes for the committee size, and does not sample the full set.
-   *          This is more efficient than the stupid way, but still not optimal.
-   *          To be more clever, we can compute the `shuffeRounds` and `pivots` separately
-   *          such that they get shared accross multiple indices.
-   *
-   * @param _committeeSize - The size of the committee
-   * @param _indexCount - The total number of indices
-   * @param _seed - The seed to use for shuffling
-   *
-   * @return indices - The indices of the committee
-   */
-  function computeCommitteeClever(uint256 _committeeSize, uint256 _indexCount, uint256 _seed)
-    internal
-    pure
-    returns (uint256[] memory)
-  {
-    uint256[] memory indices = new uint256[](_committeeSize);
+    uint256 upperLimit = _indexCount - 1;
 
     for (uint256 index = 0; index < _committeeSize; index++) {
-      uint256 originalIndex = computeOriginalIndex(index, _indexCount, _seed);
-      indices[index] = originalIndex;
-    }
+      uint256 sampledIndex = computeShuffledIndex(index, upperLimit, _seed);
 
-    return indices;
-  }
+      // Get index, or its swapped override
+      sampledIndices[index] = getValue(sampledIndex);
 
-  /**
-   * @notice  Computes the shuffled index
-   *
-   * @param _index - The index to shuffle
-   * @param _indexCount - The total number of indices
-   * @param _seed - The seed to use for shuffling
-   */
-  function computeShuffledIndex(uint256 _index, uint256 _indexCount, uint256 _seed)
-    internal
-    pure
-    returns (uint256)
-  {
-    require(_index < _indexCount, Errors.SampleLib__IndexOutOfBounds(_index, _indexCount));
-    uint256 rounds = computeShuffleRounds(_indexCount);
-
-    uint256 index = _index;
+      // Swap with the last index
+      setOverrideValue(sampledIndex, getValue(upperLimit));
 
-    for (uint256 currentRound = 0; currentRound < rounds; currentRound++) {
-      uint256 pivot = computePivot(_seed, currentRound, _indexCount);
-      index = computeInner(_seed, pivot, index, currentRound, _indexCount);
+      // Decrement the upper limit
+      upperLimit--;
```

Comment:
Small one here, but if you sample n or n, this would be hitting an underflow. 

But in that case, you also would not be using the last override, so you could do:


```solidity
sampledIndices[index] = getValue(sampledIndex);
if (upperLimit > 0) {
  sampledIndices[index] = getValue(sampledIndex);
  setOverrideValue(sampleIndex, getValue(upperLimit);
}
```